channel/channel.go

```diff
@@ -224,3 +225,38 @@ func (c Channel) AddSignedStates(sss []state.SignedState) bool {
 	}
 	return allOk
 }
+
+// SignAndAddPrefund signs and adds the prefund state for the channel, returning a state.SignedState suitable for sending to peers.
+func (c Channel) SignAndAddPrefund(sk *[]byte) (state.SignedState, error) {
+	return c.signAndAddSetupState(true, sk)
+}
+
+// SignAndAddPrefund signs and adds the postfund state for the channel, returning a state.SignedState suitable for sending to peers.
+func (c Channel) SignAndAddPostfund(sk *[]byte) (state.SignedState, error) {
+	return c.signAndAddSetupState(false, sk)
+}
+
+// signAndAddPrefund signs and adds a setup state for the channel, returning a state.SignedState suitable for sending to peers.
+// If the preFund is true, the setup state is the Prefund state, otherwise it is the Postfund state.
+func (c Channel) signAndAddSetupState(preFund bool, sk *[]byte) (state.SignedState, error) {
```

Comment:
We could change this to `signAndAddState(state.State s, sk *[]byte)`  and then just call `c.signAndAddState(c.Pre/PostFundState(),sk)` in `SignAndAddPre/Postfund`.  This would have us having to use a boolean prefund flag.

Reply:
[8dc0308](https://github.com/statechannels/go-nitro/pull/211/commits/8dc03088632a1d94669808ec981bc473c9273356)